### DISCUSSION TITLE
The fields are not generated in MongoDB Database

### DISCUSSION THREAD

**Discussion by NEARworld** [prisma/prisma#22597] [Q&A]
### Question

I am using MongoDB database and prisma.

```prisma
model User {
  id             String    @id @default(auto()) @map("_id") @db.ObjectId
  name           String?
  email          String?   @unique
  emailVerified  DateTime?
  image          String?
  hashedPassword String?
  createdAt      DateTime  @default(now())
  updatedAt      DateTime  @updatedAt

  conversationIds String[]       @db.ObjectId
  conversations   Conversation[] @relation(fields: [conversationIds], references: [id])

  seenMessageIds String[] @db.ObjectId
  // seenMessage    Message[] @relation("Seen", fields: [seenMessageIds], references: [id])

  accounts Account[]
  // messages Message[]
}
```
for example, `conversationIds` `conversations` `accounts` aren't optional fields.
So I thought they have to be in database when I create a user in the database.

<img width="603" alt="image" src="https://github.com/prisma/prisma/assets/102969108/cc3d8b09-a2be-447b-9915-1d2f1a30f14c">

but I didn't find anything those fields in the database.

**Comment by ludralph** [comment]
Hi @NEARworld 👋 

Thank you for raising this question.

Relation fields `conversations` and `accounts` define connections between models at the Prisma level and do not exist in the database. These fields are used to generate the Prisma Client. See the [documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations)

 The scalar field `conversationId`, which is referenced by the @relation attribute does exist in the database because it is the foreign key that connects `Users` and `Conversations`. Can you please share the complete schema that shows the `Conversation` and `Account` model as well to understand why `conversationId isn't shown?

**Reply by NEARworld** [reply]
Hi, thanks for your answer. here's my User and Conversation models.
```tsx
model User {
  id             String    @id @default(auto()) @map("_id") @db.ObjectId
  name           String?
  email          String?   @unique
  emailVerified  DateTime?
  image          String?
  hashedPassword String?
  createdAt      DateTime  @default(now())
  updatedAt      DateTime  @updatedAt

  conversationIds String[]       @db.ObjectId
  conversations   Conversation[] @relation(fields: [conversationIds], references: [id])

  seenMessageIds String[]  @db.ObjectId
  seenMessage    Message[] @relation("Seen", fields: [seenMessageIds], references: [id])

  accounts Account[]
  messages Message[]
}
```
```tsx
model Conversation {
  id            String   @id @default(auto()) @map("_id") @db.ObjectId
  createdAt     DateTime @default(now())
  lastMessageAt DateTime @default(now())
  name          String?
  isGroup       Boolean?

  messageIds String[]  @db.ObjectId
  messages   Message[]

  userIds String[] @db.ObjectId
  users   User[]   @relation(fields: [userIds], references: [id])
}
```

**Reply by ludralph** [reply]
Hi @NEARworld, Thank you for sharing your schema.

Using this stripped down schema between User and Conversation,
```prisma
model User {
  id             String    @id @default(auto()) @map("_id") @db.ObjectId
  name           String?
  email          String?   @unique
  emailVerified  DateTime?
  image          String?
  hashedPassword String?
  createdAt      DateTime  @default(now())
  updatedAt      DateTime  @updatedAt

  conversationIds String[]       @db.ObjectId
  conversations   Conversation[] @relation(fields: [conversationIds], references: [id])
}
model Conversation {
  id            String   @id @default(auto()) @map("_id") @db.ObjectId
  createdAt     DateTime @default(now())
  lastMessageAt DateTime @default(now())
  name          String?
  isGroup       Boolean?

  userIds String[] @db.ObjectId
  users   User[]   @relation(fields: [userIds], references: [id])
}
```
The scalar field `conversationIds` is present in the underlying database.
<img width="402" alt="image" src="https://github.com/prisma/prisma/assets/13855608/f747d200-f8ba-46b8-9f01-271446b2c4f0">

To reiterate, the relation field `conversations` is used to generate Prisma Client and will not exist in the database, but the scalar field `conversationIds` should exist.  Can you please share the query you are performing?

**Reply by NEARworld** [reply]
```tsx
    const existingConversations = await prisma.conversation.findMany({
      where: {
        OR: [
          {
            userIds: {
              equals: [currentUser.id, userId],
            },
          },
          {
            userIds: {
              equals: [userId, currentUser.id],
            },
          },
        ],
      },
    });
```
Here's the query. I will also put the link for the file content just in case.
https://github.com/NEARworld/next-messenger/blob/main/app/api/conversations/route.ts

**Reply by ludralph** [reply]
Hi @NEARworld 

What is the output of this query you shared? Is the `conversationIds` not included in the resultset?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
